ivy/functional/frontends/torch/nn/functional/vision_functions.py

```diff
@@ -120,3 +120,108 @@ def pad(input, pad, mode="constant", value=0):
                 + "['constant', 'reflect', 'replicate', 'circular']"
             ).format(mode)
         )
+
+
+@to_ivy_arrays_and_back
+def upsample_bilinear(input, size=None, scale_factor=None):
+    if scale_factor and size:
+        raise ivy.exceptions.IvyException(
+            ("only one of size or scale_factor should be defined")
+        )
+    elif (not scale_factor) and (not size):
+        raise ivy.exceptions.IvyException(
+            ("either size or scale_factor should be defined")
+        )
+
+    n, c, w_old, h_old = input.shape
+
+    if scale_factor and (not size):
+        w_new = int(w_old * scale_factor)
+        h_new = int(h_old * scale_factor)
+    elif (not scale_factor) and size:
+        w_new, h_new = size
```

Comment:
From my understanding, you are assuming that the size argument is always a tuple, and the scale_factor always an int. From reading the docs for [torch.nn.functional.upsample_bilinear](https://pytorch.org/docs/stable/generated/torch.nn.functional.upsample_bilinear.html) i see that the size argument can also be an int, while the scale_factor can also be a tuple, you may want to take this into consideration

Reply:
Thank you for invaluable review, should be resolved by the latest commits.